app/routines/search_users.rb

```diff
@@ -36,33 +40,51 @@ class SearchUsers
   SORT_DESCENDING = 'DESC'
   MAX_MATCHING_USERS = 10
 
+  def matches(field)
+    debugger
+    field.like_any
+  end
+
   def exec(query, options={})
 
     users = User.scoped
-    
+
     KeywordSearch.search(query) do |with|
 
-      with.default_keyword :any
+      match_method = options[:exact] ? :any : :like_any
 
-      with.keyword :username do |usernames|
-        users = users.where{username.like_any my{prep_names(usernames)}}
-      end
+      compose_clause = if options[:exact]
+                         lambda{ |q,field,names| q.lower(field).any prep_names(names) }
+                       else
+                         lambda{ |q,field,names| q.lower(field).like_any prep_names(names) }
+                       end
 
       with.keyword :first_name do |first_names|
-        users = users.where{lower(first_name).like_any my{prep_names(first_names)}}
+         users = users.where{
+           lower(first_name).send(match_method, my{prep_names(first_names)})
+         }
+      end
+
+      with.default_keyword :any
+      with.keyword :username do |usernames|
+        users = users.where{ | q | compose_clause[q, q.username, usernames] }
       end
 
       with.keyword :last_name do |last_names|
-        users = users.where{lower(last_name).like_any my{prep_names(last_names)}}
+        users = users.where{lower(last_name).method(match_method).call my{prep_names(last_names)}}
       end
 
       with.keyword :full_name do |full_names|
-        users = users.where{lower(full_name).like_any my{prep_names(full_names)}}
+        users = users.where{
+          options[:exact] ?
+            lower(full_name).any( my{prep_names(full_names)} ) :
+            lower(full_name).like_any( my{prep_names(full_names)} )
+        }
       end
```

Comment:
This is the "naive" implementation where it would just check the options on every block.  I think it would get pretty messy